File: javascript/easy/assignCookies.js
/**
 * @param {number[]} g
 * @param {number[]} s
 * @return {number}
 */
var findContentChildren = function(g, s) {
    g.sort(function(x,y){
        return x-y;
    });
    s.sort(function(x,y){
        return x-y;
    });
    let i = 0, j = 0;
    while(i < g.length && j < s.length) {
        if(s[j] >= g[i]) {
            i++;
        }
        j++;
    }
    return i;
};

File: ruby/easy/assign_cookies.rb
# @param {Integer[]} g
# @param {Integer[]} s
# @return {Integer}
def find_content_children(g, s)
    g.sort!
    s.sort!
    i, j = 0, 0
    while i < g.length && j < s.length
        i += 1 if g[i] <= s[j]
        j += 1
    end
    return i
end
